client/pruning/main.go

```diff
@@ -21,7 +21,7 @@ const FlagAppDBBackend = "app-db-backend"
 
 // Cmd prunes the sdk root multi store history versions based on the pruning options
 // specified by command flags.
-func Cmd(appCreator servertypes.AppCreator, defaultNodeHome string) *cobra.Command {
+func Cmd[T servertypes.Application](appCreator servertypes.AppCreator[T], defaultNodeHome string) *cobra.Command {
```

Comment:
It looks like the only diff in this PR is the generics.  If that is the case, then we can just close

Reply:
I mean the purpose of this pr is to allow calling back into the application struct in PostSetup. If that something we don't want to support in the future for berachain then we can close this.

Reply:
the PR extends the StartCmdOptions struct here: https://github.com/cosmos/cosmos-sdk/pull/24047/files#diff-97289af2ab66016f168df6fccac70626d125f170e5cf0841fcdc8fa624999608R112-R124

the generics help with type safety i believe - so you cannot pass in an AppCreator for a different type than your application.